File: app/src/main/java/com/ddiehl/android/htn/view/FragmentActivity.java
package com.ddiehl.android.htn.view;

import android.app.Fragment;

import com.ddiehl.android.htn.R;

public abstract class FragmentActivity extends BaseActivity {

    protected abstract Fragment getFragment();

    protected abstract String getFragmentTag();

    @Override
    public void onStart() {
        super.onStart();
        if (getFragmentManager().findFragmentByTag(getFragmentTag()) == null) {
            getFragmentManager().beginTransaction()
                    .add(R.id.fragment_container, getFragment(), getFragmentTag())
                    .commit();
        }
    }
}

File: app/src/main/java/com/ddiehl/android/htn/view/FragmentActivity.kt
package com.ddiehl.android.htn.view

import android.app.Fragment
import com.ddiehl.android.htn.R

@Suppress("DEPRECATION") // no kidding
abstract class FragmentActivity : BaseDaggerActivity() {

    protected abstract val fragment: Fragment?

    protected abstract val fragmentTag: String?

    public override fun onStart() {
        super.onStart()
        if (fragmentManager.findFragmentByTag(fragmentTag) == null) {
            fragmentManager.beginTransaction()
                .add(R.id.fragment_container, fragment, fragmentTag)
                .commit()
        }
    }
}
